### PR TITLE
Add new test CodecoolerDAO

### PR DESCRIPTION
**Pull request by idzia**

### CODE REVIEW

File: src/main/java/DAO/CodecoolerDAO.java [hunk 1]
```diff
@@ -21,6 +21,10 @@ public CodecoolerDAO(Connection connection, InventoryDAO inventoryDAO) {
         this.inventoryDAO = inventoryDAO;
     }
 
+//    public CodecoolerDAO(Connection connection) {
+//        this.connection = connection;
+//        this.inventoryDAO = ;
+//    }
```
Comment (kret0) [inline src/main/java/DAO/CodecoolerDAO.java]: dont leave comments

File: src/main/java/DAO/FactoryDAO.java [hunk 2]
```diff
@@ -27,10 +27,13 @@ public QuestDAO getQuestDAO() {
...
     public CodecoolerDAO getCodecoolerDAO() {
-        return new CodecoolerDAO(createConnection(), new InventoryDAO());
+        return new CodecoolerDAO(createConnection(), getInventoryDAO());
     }
     public MentorDAO getMentorDAO() {
         return new MentorDAO(createConnection());
     }
 
+    public InventoryDAO getInventoryDAO() {
+        return new InventoryDAO(createConnection());
```
Comment (kret0) [inline src/main/java/DAO/FactoryDAO.java]: it shoudl be protected at least

File: src/main/java/DAO/InventoryDAO.java [hunk 3]
```diff
@@ -13,9 +13,15 @@
...
+    private Connection connection;
+
+    public InventoryDAO(Connection connection){
+        this.connection = connection;
+    }
+
     public void addArtefactToInventory(Artifact artifact, Codecooler codecooler) throws SQLException {
 
-        Connection connection = DatabaseConnection.getInstance().getConnection();
+//        Connection connection = DatabaseConnection.getInstance().getConnection();
```
Comment (kret0) [inline src/main/java/DAO/InventoryDAO.java]: don't leave comments cdn

File: src/main/java/DAO/InventoryDAO.java [hunk 4]
```diff
@@ -34,7 +40,7 @@ public void addArtefactToInventory(Artifact artifact, Codecooler codecooler) thr
         List<Integer> artefactIdList = new ArrayList<>();
         System.out.println(codecoolerId);
 
-        Connection connection = DatabaseConnection.getInstance().getConnection();
+//        Connection connection = DatabaseConnection.getInstance().getConnection();
```
Comment (kret0) [inline src/main/java/DAO/InventoryDAO.java]: don't leave comments cdn

File: src/main/java/DAO/InventoryDAO.java [hunk 5]
```diff
@@ -55,7 +61,7 @@ public void addArtefactToInventory(Artifact artifact, Codecooler codecooler) thr
     public ArrayList<Artifact> getListOfArtifact(List<Integer> artifactIdList) throws SQLException {
         ArrayList<Artifact> artifactList = new ArrayList<>();
 
-        Connection connection = DatabaseConnection.getInstance().getConnection();
+//        Connection connection = DatabaseConnection.getInstance().getConnection();
```
Comment (kret0) [inline src/main/java/DAO/InventoryDAO.java]: don't leave comments cdn

File: src/test/DAO/CodecoolerDAOTest.java [hunk 6]
```diff
@@ -44,7 +44,7 @@ public void setUp() throws SQLException {
         when(preparedStatement.executeQuery()).thenReturn(resultSet);
 
         codecoolerDAO = new CodecoolerDAO(connection, inventoryDAO);
-
+//        codecoolerDAO = new CodecoolerDAO(connection);
```
Comment (kret0) [inline src/test/DAO/CodecoolerDAOTest.java]: ...

File: src/test/DAO/CodecoolerDAOTest.java [hunk 7]
```diff
@@ -102,4 +102,48 @@ public void testCodecoolerIdByNotExistUser() throws SQLException {
...
+        assertEquals(codecoolerList.get(0).getPassword(), expectedCodecoolerList.get(0).getPassword());
+        assertEquals(codecoolerList.get(0).getEmail(), expectedCodecoolerList.get(0).getEmail());
+        int experience = codecoolerList.get(0).getExp();
+        assertEquals(experience, 0);
+        assertNull(codecoolerList.get(0).getRoomName());
+        assertNull(codecoolerList.get(0).getInventory());
+    }
+
+    @Test
+    public void testGetListOfCodecoolersWhenCodecoolersNotExist() throws SQLException {
```
Comment (kret0) [inline src/test/DAO/CodecoolerDAOTest.java]: as @feniXb3 suggested, it is nice, when tests sounds like 
testGettingListOfCodecoolersWhenCodecoolersNotExist()

**Comment by idzia** [comment]
is done

**Pull request merged**